File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        int i = 0;
        for (int num : nums) {
            if (mp.contains(target - num)) {
                return {i, mp[target - num]};
            }
            mp[num] = i;
            i++;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in mp:
                return [i, mp[complement]]
            mp[num] = i
        return []